File: leetsync/1895-minimum-number-of-operations-to-move-all-balls-to-each-box/minimum-number-of-operations-to-move-all-balls-to-each-box.java
class Solution {
    public int[] minOperations(String boxes) {
        int n = boxes.length();
        int[] answer = new int[n];
        int cumValue = 0, cumValueSum = 0;
        // Find moves for all left balls to index i
        for (int i = 0; i < n; i++) {
            answer[i] = cumValueSum;
            cumValue += boxes.charAt(i) == '0' ? 0 : 1;
            cumValueSum += cumValue;
        }
        cumValue = 0;
        cumValueSum = 0;
        // Find moves for all right balls to index i
        for(int i=n-1; i>=0; i--){
            answer[i]+=cumValueSum;
            cumValue+=boxes.charAt(i)=='0'?0:1;
            cumValueSum+=cumValue;
        }
        return answer;
    }
}

File: leetsync/1895-minimum-number-of-operations-to-move-all-balls-to-each-box/minimum-number-of-operations-to-move-all-balls-to-each-box.go
func minOperations(boxes string) []int {
	n := len(boxes)
	answer := make([]int, n)
	cumValue, cumValueSum := 0, 0
	for i := 0; i < n; i++ {
		answer[i] = cumValueSum
		cumValue += int(boxes[i] - '0')
		cumValueSum += cumValue
	}
	cumValue, cumValueSum = 0, 0
	for i := n - 1; i >= 0; i-- {
		answer[i] += cumValueSum
		cumValue += int(boxes[i] - '0')
		cumValueSum += cumValue
	}
	return answer
}